lib/std/zig/parse.zig

```diff
@@ -48,29 +48,32 @@ pub fn parse(allocator: *Allocator, source: []const u8) !*Tree {
 
     while (it.peek().?.id == .LineComment) _ = it.next();
 
-    tree.root_node = try parseRoot(arena, &it, tree);
+    tree.root_node = parseRoot(arena, &it, tree) catch |err| blk: {
+        switch (err) {
+            error.ParseError => {
+                assert(tree.errors.len != 0);
+                break :blk undefined;
+            },
+            error.OutOfMemory => {
+                return error.OutOfMemory;
+            },
+        }
+    };
+
```

Comment:
I think what you have here is reasonable, but I'd prefer to revert this to what it was previously. This way API users are less likely to forget to check for the error. So when the user gets `error.ParseError` then they know to check for errors.

The "abuse of error.ParseError" that I mentioned earlier, I just meant the other thing that you already fixed on line 65.

Reply:
I don't understand what you mean. The API did not return `error.ParseError` previously.

Reply:
My mistake- I failed to notice the previous error set of `parseRoot` was `Allocator.Error`. I mistakenly thought you had changed the error set of `parse` to no longer include `error.ParseError`.

In this case I take back what I said, and I think this is good to merge :+1: